### ISSUE TITLE
Feature Request, keep the scroll position when inserting new items on top of SuperListView.

### ISSUE THREAD

**Issue by ErinReade** [superlistapp/super_sliver_list#53]
Awesome work, I've tried lots of similar packages, SuperListView is the killer! 
I'd like to contribute to it if you have such a plan.
Thank you for your outstanding work.

**Comment by jpohhhh** [comment]
I'd _love_ this as well. I'd be willing to sponsor it, especially if it could be done in the short term.

TL;DR:
I'm building an app that integrates with several LLM APIs. When a new AI message streams in, the user is "autoscrolled" to the bottom of the new message. [video](https://github.com/superlistapp/super_sliver_list/assets/65884233/72a2b798-f9b9-4b0f-a295-9c353d2718f0)

The LLMs are fast enough now that I find it extremely disruptive: your eye is drawn to speed-read it. You miss half of it, even though it it feels like you read it, and to start from the beginning, you have to wait till the end, then manually scroll up.


Commentary:
There's an odd gap in Flutter currently where you can't really build a "chat" UI that doesn't autoscroll. As far as I can tell, after months of trying everything: 
- you need to do ListView(reversed: true) to get "chat starts at most recent message when user navigates to it"
- if you do that, then when a new message comes in, the list will "scroll down" by height of the new message. i.e. if I'm at the bottom, I see the whole message come in, but if I'm scrolled up by X, I see new message - X.
- the most touted solution for this is `PositionRetainedScrollPhysics`. It doesn't seem to work in the reversed case at all. When I tried them in the non-reversed case, they had the classic "off by one frame" problem, where you see a jump animation.


Btw can't thank you enough for this package. I accidentally fixed the version to 0.0.7, and even 0.07  was a massive leap forward from the behavior with standard ListView. After upgrading to 0.4.1, I'm frankly stunned, I don't know how you got it to be so smooth in quite challenging circumstances - the tiles I render are in Markdown and of quite varying length, and there's a variety of tiles, chat, art, rows with search results.

**Comment by knopp** [comment]
Is reverse list necessary here? Wouldn't regular list where you just jump at the last item initially work here?
